xformers/ops.py

```diff
@@ -655,6 +655,46 @@ def from_arguments(
         )
 
 
+class Chunk3(torch.autograd.Function):
+    """
+    Splits a packed `qkv` tensor into query, key and values.
+    The magic happens in the backward. We want to `torch.stack` the tensors
+    together, but we don't need to if the gradients have already the same storage
+    (and that is something that our attention operators support)
+    """
+
+    @staticmethod
+    # type: ignore
+    def forward(ctx, qkv: torch.Tensor, dim: int):
+        q, k, v = qkv.select(dim, 0), qkv.select(dim, 1), qkv.select(dim, 2)
```

Comment:
nit: you can make the code a bit more generic by doing `qkv.unbind(dim=dim)`. This way you can support arbitrary number of elements in the dimension, as you don't currently have a check here that there are exactly 3 elements.

If you use `unbind`, it might make sense to call this function `efficient_unbind` or something like that